Добавил модели Question и Answer, написал тесты к ним

Tekkon opened the pull request.

Commit:
Добавил модели Question и Answer, написал тесты к ним

Commit:
1. Удалил поле title у Answer
2. Добавил связь между Answer и Question и тест этой связи

vkurennov commented on app/models/answer.rb:
```diff
@@ -0,0 +1,5 @@
+class Answer < ApplicationRecord
+  validates :body, presence: true
```
В моделях принят порядок: сначала ассоциации, затем - валидации, придерживайся такого же порядка

vkurennov commented on app/models/question.rb:
```diff
@@ -0,0 +1,5 @@
+class Question < ApplicationRecord
+  validates :title, :body, presence: true
+
+  has_many :answers
```
* Тут нужно добавить опцию, чтобы при удалении вопроса удалялись и связанные с ним ответы. И не забудь указать это в тестах, shoulda-matchers умеет это делать (отсутствует dependent: :destroy)

vkurennov commented on config/database.yml:
```diff
@@ -1,25 +1,14 @@
-  adapter: sqlite3
+  adapter: postgresql
+  username: postgres
+  password: pwgqsyx2
```
Не стоило включать database.yml в репозиторий, тем более с паролем. Надо добавить в .gitignore его и удалить из репозитория, а локально сделать нужную копию файла из database.yml.sample. Ведь у разных разработчиков могут быть разные локальные настройки, да и пароли светить небезопасно

vkurennov commented on db/migrate/20170811111522_add_title_and_body_to_question.rb:
```diff
@@ -0,0 +1,6 @@
+class AddTitleAndBodyToQuestion < ActiveRecord::Migration[5.0]
+  def change
+  	add_column :questions, :title, :string
```
Пока не слил ветку в мастер, лучше откатить миграцию и исправить ее.

vkurennov commented on db/migrate/20170811142251_add_question_id_to_answer.rb:
```diff
@@ -0,0 +1,5 @@
+class AddQuestionIdToAnswer < ActiveRecord::Migration[5.0]
+  def change
+  	add_column :answers, :question_id, :integer
```
Лучше использовать add_belongs_to, т.к. в этом случае сразу индекс на поле будет добавлен

vkurennov commented on spec/models/answer_spec.rb:
```diff
@@ -0,0 +1,6 @@
+
+RSpec.describe Answer, type: :model do
+  it { should validate_presence_of :body }
+  it { should belong_to(:question) }  
```
В тестах тоже придерживайся порядка "снчала ассоциации, потом - валидации".

Commit:
1. В моделях перенес описание ассоциаций перед описанием валидаций
2. В Question добавил опцию dependent: :destroy, чтобы связанные с вопросом ответы удалялись автоматически при удалении вопроса.
3. Исключил database.yml из репозитория
4. Исправил миграции
5. Изменил порядок тестов: сначала ассоциации, потом валидации
